### DISCUSSION TITLE
RedisClient issue: getting a NullPointerException

### DISCUSSION THREAD

**Discussion by flowt-au** [quarkusio/quarkus#35269] [Q&A]
I am using Quarkus 2.16.9.Final and the `quarkus-redis-client`. I am using the RedisJson commands as shown below.

The issue is if I do a jsonGet() for a document that does not exist, Redis correctly returns `null` but I get 
```
java.lang.NullPointerException: Cannot invoke "io.vertx.mutiny.redis.client.Response.toBuffer()" because "r" is null
```
i.e. it looks like a bug in the vertx client.

My question is how do i find out what version of that package I am using? I am reluctant to upgrade to Quarkus V3 right now because I have enough backlog to keep me busy. But, if it turns out this has been fixed in a later version and is in Quarkus v3 I will bite the bullet and upgrade. Or, is it possible to use the latest Vertx Redis Client with Quarkus 2.16.9.Final? I am not experienced enough in Java yet to know how to do that (even if it is possible).

```
// Abbreviated setup:
import io.quarkus.redis.datasource.json.ReactiveJsonCommands;
protected ReactiveJsonCommands<String> jsonClient;
this.jsonClient = reactiveRedisDS.json();

return jsonClient.jsonGet(redisdbkey + ":" + mkey, "$")
    .onItem()
    .transformToUni(jsonResultArr -> {
      // It only reaches here if the document exists in Redis.

      // System.out.println("transform response jsonResultArr=" + jsonResultArr.toString());
      // Make a Uni ...
    });
```
Thanks,
Murray

[a comment by an automated account is not shown]

**Reply by cescoffier** [reply]
Yes, it got fixed in Quarkus 3.0

**Comment by flowt-au** [comment]
Ah ha... thank you. :-)
